tools/upload_to_aws.py

```diff
@@ -17,40 +21,52 @@
 idx_file_name = task_dict.get('input').get('idx_file_name')
 idx_object_id = task_dict.get('input').get('idx_object_id')
 idx_file_md5sum = task_dict.get('input').get('idx_file_md5sum')
+project_code = task_dict.get('input').get('project_code')
+
 
-if file_md5sum is None:
-    file_md5sum = str(get_md5(file_))
+task_start = int(time.time())
+file_size = 0
 
+if project_code in allowed_codes:
+    if file_md5sum is None:
+        file_md5sum = str(get_md5(file_))
 
-if idx_file_ and idx_file_md5sum is None:
-    idx_file_md5sum = str(get_md5(idx_file_))
 
-file_size = int(os.path.getsize(file_))
+    if idx_file_ and idx_file_md5sum is None:
+        idx_file_md5sum = str(get_md5(idx_file_))if file_md5sum is None:
+            file_md5sum = str(get_md5(file_))
 
+    if idx_file_ and idx_file_md5sum is None:
+        idx_file_md5sum = str(get_md5(idx_file_))
 
-task_start = int(time.time())
+    task_info= 'Project code %s is allowed. Starting Upload.' % project_code
+    file_size = int(os.path.getsize(file_))
+
+    if idx_object_id:
+        file_size+= + int(os.path.getsize(idx_file_))
+        try:
+            print subprocess.check_output(['icgc-storage-client','upload','--file', idx_file_, '--object-id', idx_object_id, '--md5', idx_file_md5sum, '--force'])
```

Comment:
The default of `--profile` is `aws` for `icgc-storage-client` , that's why for `upload_to_aws`, the configuration is omitted.